### PR TITLE
New Tool: OMERO metadata upload using ezomero

### PR DESCRIPTION
**Pull request by rmassei**
**New Tool to upload metadata in a user-defined OMERO.server using [ezomero](https://github.com/TheJacksonLaboratory/ezomero)**

- The tool is linking tables and key-value pairs to an existing OMERO object (project, dataset, screen or image) by input its ID. Otherwise, a new object will be create and it will be named as the importing date and time.
- Input is tsv file. For Key-Value pairs, Headers are keys and each row is a value.
- Output is a txt report with the object ID where metadata where uploaded

Furthermore, this pull request re-order in specific folders the existing omero tools.

### CODE REVIEW

**Comment by rmassei** [comment]
https://github.com/Helmholtz-UFZ/galaxy-tools/actions/runs/10178032591?pr=36
It seems that the metadata where uploaded successfully,

Output log:  different than expected, difference (using diff):
( /tmp/tmp_5xmjjiaoutput_table_import.txt v. /tmp/tmpxonjhmmsoutput_table_import.txt )
--- local_file
+++ history_data
@@ -1,2 +1,2 @@
-2024-07-31 11:16:08.233267 - SUCCESS: Successfully uploaded metadata for project with ID 1. Result:      Key1    Key2
+2024-07-31 10:05:18.338433 - SUCCESS: Successfully uploaded metadata for project with ID 1. Result:      Key1    Key2
 0  Value1  Value2

but maybe there is a problem in assessing the log file content @bernt-matthias?

**Comment by bernt-matthias** [comment]
How do I need to interpret the diff? I do not really understand what the problem is.

**Comment by rmassei** [comment]
There was an issue with the log output file; the timestamps prevented the files from being comparable. :)

File: tools/omero/omero_metadata_upload/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+categories:
```
Comment (bernt-matthias) [inline tools/omero/omero_metadata_upload/.shed.yml:1]: Can you move  all omero related tools to `tools/omero/` and use the dhed.yaml of the upload tool

File: tools/omero/omero_metadata_upload/omero_metadata_import.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        </when>
+        <when value="existing_object">
+            <param name="did" type="integer" value="1409"/>
+        </when>
+        </conditional>
+        <param argument="ann_type" type="select" optional="false" label="Annotation type" help="Select annotation format">
+            <option value="table">An OMERO Table</option>
+            <option value="KV">OMERO Key-Value Pairs</option>
+        </param>
+        <param argument="ann_file" type="data" format="tabular" optional="false" label="Annotation file" help="Select annotation file"/>
```
Comment (bernt-matthias) [inline tools/omero/omero_metadata_upload/omero_metadata_import.xml:55]: Can you document how such a table looks.

File: tools/omero/omero_metadata_upload/.shed.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- Imaging
+name: omero_metadata_upload
+description: Add metadata to an object stored in an OMERO.server using ezomero
+long_description: Create project, dataset, screen and/or attach metadata to an object stored in an OMERO.server
+owner: ufz
+remote_repository_url: https://github.com/Helmholtz-UFZ/galaxy-tools/tree/main/tools/omero/omero_metadata_upload
+homepage_url: https://github.com/TheJacksonLaboratory/ezomero
+auto_tool_repositories:
```
Comment (bernt-matthias) [inline tools/omero/omero_metadata_upload/.shed.yml:9]: `auto_tool_repositories` only makes sense with all tools in one dir. Then we do not need to create separate shed.yaml files for each of them. 

In the end it's just a personal preference. If you like this more .. also fine.

Comment (rmassei) [reply tools/omero/omero_metadata_upload/.shed.yml:9]: No problem, it helps to keep everything more organized if we put all omero tools in one dir. I will take care to put everything in one folder and update the shed.yml (https://github.com/Helmholtz-UFZ/galaxy-tools/pull/36/files#r1699114013)

File: tools/omero/omero_metadata_upload/omero_metadata_import.xml (hunk 2 above)
Comment (rmassei) [reply tools/omero/omero_metadata_upload/omero_metadata_import.xml]: Done, I had a help section in the .xml

File: tools/omero/omero_metadata_upload/.shed.yml (hunk 1 above)
Comment (rmassei) [reply tools/omero/omero_metadata_upload/.shed.yml]: https://github.com/Helmholtz-UFZ/galaxy-tools/pull/36#discussion_r1699532878

**Comment by rmassei** [comment]
Ready to go on my side :smiley:

**Review by rmassei — COMMENTED** [review]
Reviewed the last changes, thanks for this!

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**